src/Core/src/Credentials/ConfigurationProvider.php

```diff
@@ -23,7 +23,9 @@ public function getCredentials(Configuration $configuration): ?Credentials
             return new Credentials(
                 $configuration->get(Configuration::OPTION_ACCESS_KEY_ID),
                 $configuration->get(Configuration::OPTION_SECRET_ACCESS_KEY),
-                $configuration->get(Configuration::OPTION_SESSION_TOKEN)
+                $configuration->get(Configuration::OPTION_SESSION_TOKEN),
+                null,
```

Comment:
What expire date will be used here?

Reply:
Should we even cache this BTW? It is quicker to read from memory that to use any cache

Reply:
For AWS when null: never expires.
When using our new CacheProvider, it uses the CacheAdapter default lifetime.

> It is quicker to read from memory that to use any cache

Yes, we may avoid storing theme for nothing. But when decorating the ChainProvider with a cache layer, the cache will always be called for read.